test/e2e/podAffinityAntiAffinity.go

```diff
@@ -0,0 +1,710 @@
+				},
+			},
+		}
+
+		By("Trying to launch the setup podgroup one with label security S1")
+		_, setupPgOne := createJob(context, setupPodGrpOne)
+
+		By("validate if setup podgroup is running and get the node of the setup podgroup")
+		err := waitPodGroupReady(context, setupPgOne)
+		checkError(context, err)
+
+		nodenames := getNodesOfPodGroup(context, setupPgOne)
+
+		// this should be a single nodename
+		By("Check if the Podgroup is running on a single node, since it has only one task")
+		Expect(len(nodenames)).To(Equal(int(1)))
+
+		By(fmt.Sprintf("Setup PodGroup is running on %s", nodenames[0]))
+		setupPodGrpOnesNodeName = nodenames[0]
+
+		By("Deploy the test PodGroup with pod affinity hard constraint and check if its running")
+		_, testPg := createJob(context, testPodGrp)
+		err = waitPodGroupReady(context, testPg)
+		checkError(context, err)
+
+		By("Get the node name of the Podgroup")
+		nodenames = getNodesOfPodGroup(context, testPg)
+
+		// this should be a single nodename
+		By("Check if the Podgroup is running on a single node, since it has only one task")
+		Expect(len(nodenames)).To(Equal(int(1)))
+
+		By("validate if test PodGroup is running on the right node")
+		Expect(nodenames[0]).To(Equal(setupPodGrpOnesNodeName))
+	})
+
+	//2
+	It("validates scheduler respect's pod-anti-affinity with hard constraint", func() {
+		context := initTestContext()
+		defer cleanupTestContext(context)
+
+		podAntiAffinityHard := &v1.Affinity{
+			PodAntiAffinity: &v1.PodAntiAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+					{
+						LabelSelector: &metav1.LabelSelector{
+							MatchExpressions: []metav1.LabelSelectorRequirement{
+								{
+									Key:      "security",
+									Operator: metav1.LabelSelectorOpIn,
+									Values: []string{
+										"S1",
+									},
+								},
+							},
+						},
+						TopologyKey: "kubernetes.io/hostname",
+					},
+				},
+			},
+		}
+
+		testPodGrp := &jobSpec{
+			name: testPodGrpName,
+			tasks: []taskSpec{
+				{
+					img:      "nginx",
+					min:      1,
+					rep:      1,
+					affinity: podAntiAffinityHard,
+				},
+			},
+		}
+
+		By("Trying to launch the setup podgroup one with label security S1")
+		_, setupPgOne := createJob(context, setupPodGrpOne)
+
+		By("validate if setup podgroup is running and get the node of the setup podgroup")
+		err := waitPodGroupReady(context, setupPgOne)
+		checkError(context, err)
+
+		nodeNames := getNodesOfPodGroup(context, setupPgOne)
+
+		// this should be a single node name
+		By("Check if the Podgroup is running on a single node, since it has only one task")
+		Expect(len(nodeNames)).To(Equal(int(1)))
+
+		By(fmt.Sprintf("Setup PodGroup is running on %s", nodeNames[0]))
+		setupPodGrpOnesNodeName = nodeNames[0]
+
+		By("Deploy the test PodGroup with pod affinity hard constraint and check if its running")
+		_, testPg := createJob(context, testPodGrp)
+		err = waitPodGroupReady(context, testPg)
```

Comment:
I am not sure if the CI is running on multi-nodes cluster or one node, In one node cluster, this will fail

@k82cn

Reply:
You were right this would fail on cluster with single node. 
I have solved this issue, now these test cases will check if the test cluster has the desirable number of nodes before running a test or it will simply skip that test case.